src/js/polyfills/fetch/fetch.js

```diff
@@ -16,19 +16,23 @@ export function fetch(input, init) {
             xhr.abort();
         }
 
-        xhr.onload = function() {
-            const options = {
-                statusText: xhr.statusText,
-                headers: parseHeaders(xhr.getAllResponseHeaders() || ''),
-                status: xhr.status,
-                url: xhr.responseURL
-            };
-            const body = xhr.response;
-
-            setTimeout(function() {
-                resolve(new Response(body, options));
-            }, 0);
-        };
+        //tjs only
+        let controller=null;
+        const responseReader=new ReadableStream({
+            start(controller_){controller=controller_;}
+        });
+
+        xhr.onprogress=function(){
+            if(controller!=null){
+                controller.enqueue(new Uint8Array(xhr.__tjsGetAndClearResponseBuffer()));
```

Comment:
Let's make this function hidden under some symbol rather than a `__` name. Something like `xhr[kXhrGetAndClearResponseBuffer]()`